frontend/public/components/secret.jsx

```diff
@@ -84,21 +84,44 @@ const SecretDetails = ({obj: secret}) => {
 const SecretsList = props => <List {...props} Header={SecretHeader} Row={SecretRow} />;
 SecretsList.displayName = 'SecretsList';
 
-const secretType = secret => secret.type;
+const IMAGE_FILTER_VALUE = 'Image';
+const SOURCE_FILTER_VALUE = 'Source';
+const TLS_FILTER_VALUE = 'TLS';
+const SA_TOKEN_FILTER_VALUE = 'Service Account Token';
+const OPAQUE_FILTER_VALUE = 'Opaque';
+
+const secretTypeFilterValues = [
+  IMAGE_FILTER_VALUE,
+  SOURCE_FILTER_VALUE,
+  TLS_FILTER_VALUE,
+  SA_TOKEN_FILTER_VALUE,
+  OPAQUE_FILTER_VALUE
+];
+
+export const secretTypeFilterReducer = secret => {
+  switch (secret.type) {
+    case 'kubernetes.io/dockercfg':
```

Comment:
We already have defined k8s secret types as enum in [create-secret.tsx](https://github.com/openshift/console/blob/master/frontend/public/components/secrets/create-secret.tsx#L19-L21). 
I know that there are only three of them(wanted to add all of them as I work on the create secret form) but I would suggest to put all of the know types there and use the enum.